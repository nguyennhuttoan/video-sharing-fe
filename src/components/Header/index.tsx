import { Row, Col, Divider } from "antd";
import { Login } from "../Login";

export const Header = () => {
  return (
    <>
      <Row align="middle">
        <Col span={9} offset={3}>
          <div style={{ display: "flex" }}>
            <h1>Video Sharing</h1>
          </div>
        </Col>
        <Col span={10} offset={2}>
          <Login />
        </Col>
      </Row>
      <Divider style={{ marginTop: 5 }} />
    </>
  );
};

export default Header;
