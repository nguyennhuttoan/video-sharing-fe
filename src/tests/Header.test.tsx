import "@testing-library/jest-dom";
import ShallowRenderer from "react-test-renderer/shallow";
import Header from "../components/Header";

describe("Header", () => {
  test("should show a h1 with the text 'Video Sharing'", () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(<Header />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
