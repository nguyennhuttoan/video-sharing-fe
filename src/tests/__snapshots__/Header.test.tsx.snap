// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > should render and match the snapshot 1`] = `
<React.Fragment>
  <Row
    align="middle"
  >
    <Col
      offset={3}
      span={9}
    >
      <div
        style={
          {
            "display": "flex",
          }
        }
      >
        <h1>
          Video Sharing
        </h1>
      </div>
    </Col>
    <Col
      offset={2}
      span={10}
    >
      <Login />
    </Col>
  </Row>
  <Divider
    style={
      {
        "marginTop": 5,
      }
    }
  />
</React.Fragment>
`;

exports[`Header > should show a h1 with the text 'Video Sharing' 1`] = `
<React.Fragment>
  <Row
    align="middle"
  >
    <Col
      offset={3}
      span={9}
    >
      <div
        style={
          {
            "display": "flex",
          }
        }
      >
        <h1>
          Video Sharing
        </h1>
      </div>
    </Col>
    <Col
      offset={2}
      span={10}
    >
      <Login />
    </Col>
  </Row>
  <Divider
    style={
      {
        "marginTop": 5,
      }
    }
  />
</React.Fragment>
`;
